### PR TITLE
formatter: export functionality improvements

### PR DESCRIPTION
**Pull request by ioannistsanaktsidis**
* Addresses #459.

Signed-off-by: Ioannis Tsanaktsidis <ioannis.tsanaktsidis@cern.ch>

### CODE REVIEW

**Comment by jmartinm** [comment]
Some small things I see:

*  [ ] Accessing the page with http://localhost:4000/search?p=&cc=HEP&of=tlcv breaks the page formatting
screen shot 2015-11-05 at 2 35 20 pm

* [ ] It would be nice if the formats received the separator `\n` or `<br/>` as a parameter in `__init__()` so that you could use the same code both in HTML and in text (on Download)

* [ ] When selecting a few search results it would be good if the top `select-all` checkbox became 'half selected' like here:

screen shot 2015-11-05 at 2 40 18 pm

To easily deselect results.

* [ ] Instead of separating `obj['format']` and `obj['prefix']` isn't it simpler to just have `obj['filename']` ?

**Comment by ioannistsanaktsidis** [comment]
@jmartinm  for the comment ```  Accessing the page with http://localhost:4000/search?p=&cc=HEP&of=tlcv breaks the page formatting ```
I fixed for now not to break, but we have to decide if we allow users to see the results with of=param then I guess we will have to apply the checkboxes functionality to that templates too. Right?

**Comment by ioannistsanaktsidis** [comment]
With the latest update:
I have just left the second bullet and want to check some staff on the 1st. The last 2 bullets are ok now. Cheers.

**Comment by ioannistsanaktsidis** [comment]
@jmartinm all bullets have been updated with the latest amend. Cheers :dancer:

**Comment by jmartinm** [comment]
Please rebase with latest `master`.

File: inspire/base/static/js/search/search_results.js [hunk 1]
```diff
@@ -133,52 +134,70 @@ require(['jquery', 'bootstrap'], function($) {
...
       }).done(function() {
-        $loading.hide();
+        $('#spinner-download').hide();
         $('#download-format').text('Download');
         $('#download-format').removeClass('disabled');
       });
     }
   });
 
+  var checkboxes = document.querySelectorAll('input.checkbox-results'),
```
Comment (jmartinm) [inline inspire/base/static/js/search/search_results.js]: you moved away from jQuery and to vanilla JavaScript?

File: inspire/base/static/js/search/search_results.js [hunk 2]
```diff
@@ -118,13 +119,13 @@ require(['jquery', 'bootstrap'], function($) {
...
 
-  var $loading = $('#spinner-download').hide();
   $("#download-format").on('click', function() {
     var obj = get_download_information($('#dropdown-export').text().trim());
     if (sList.length != 0) {
-      $loading.show();
+      $('#spinner-download').show();
       $('#download-format').text('Downloading...');
       $('#download-format').addClass('disabled');
+      console.log(sList);
```
Comment (jmartinm) [inline inspire/base/static/js/search/search_results.js]: leftover

**Comment by ioannistsanaktsidis** [comment]
@jmartinm the commit is rebased now. Concerning the indeterminate state in the select all check-box there is a problem when you select all the results one by one. The checkbox becomes checked but there are some staff missing. It will take me some time to deal with it, so should I take care or continue in the facets?

**Comment by jmartinm** [comment]
Merged 835286c94658cd151bbdd2ce21dad85955239ddb

**Pull request closed without merging**